diff --git a/scripts/build.py b/scripts/build.py
index 8d3ec5ae34..0fa59ff317 100755
--- a/scripts/build.py
+++ b/scripts/build.py
@@ -1010,6 +1010,9 @@ class Configuration(object):
         if self.target_platform in ('x86_64-macos','arm64-macos','arm64-ios','x86_64-ios') and 'linux' == sys.platform:
             strip = os.path.join(sdkfolder, 'linux', sdk.PACKAGES_LINUX_CLANG, 'bin', 'x86_64-apple-darwin19-strip')
 
+        if self.target_platform == 'arm64-linux' and self.host != self.target_platform:
+            strip = '/usr/bin/aarch64-linux-gnu-strip'
+
         run.shell_command("%s %s" % (strip, path))
         return True
