diff --git a/build_tools/waf_dynamo.py b/build_tools/waf_dynamo.py
index 4bd9ca09f7..41ce2ecf69 100644
--- a/build_tools/waf_dynamo.py
+++ b/build_tools/waf_dynamo.py
@@ -1449,6 +1449,9 @@ def run_tests(ctx, valgrind = False, configfile = None):
             if 'web' in ctx.env.PLATFORM: # should be moved to TEST_LAUNCH_ARGS
                 cmd = '%s %s' % (ctx.env['NODEJS'][0], cmd)
 
+            if ctx.env.PLATFORM != ctx.env.BUILD_PLATFORM and 'arm64-linux' in ctx.env.PLATFORM:
+                cmd = '%s %s' % (ctx.env['QEMU_AARCH64'], cmd)
+
         # disable shortly during beta release, due to issue with jctest + test_gui
         valgrind = False
         if valgrind:
@@ -1731,6 +1734,15 @@ def detect(conf):
             conf.env.CC = "clang"
             conf.env.CPP = "clang -E"
 
+        if platform == 'arm64-linux' and platform != host_platform and not Options.options.skip_tests:
+            # We use qemu to run tests on non-arm64 platforms
+            if not conf.env['QEMU_AARCH64']:
+                conf.find_program('qemu-aarch64-static', var='QEMU_AARCH64', mandatory = False)
+                if not conf.env['QEMU_AARCH64']:
+                    conf.find_program('qemu-aarch64', var='QEMU_AARCH64', mandatory = True)
+                if conf.env['QEMU_AARCH64']:
+                    conf.env['QEMU_AARCH64'] = conf.env['QEMU_AARCH64'][0]
+
     platform_setup_tools(conf, build_util)
 
     # jg: this whole thing is a 'dirty hack' to be able to pick up our own SDKs
diff --git a/engine/dlib/src/dlib/linux/socket_linux.cpp b/engine/dlib/src/dlib/linux/socket_linux.cpp
index c99d8717a6..471cca12c2 100644
--- a/engine/dlib/src/dlib/linux/socket_linux.cpp
+++ b/engine/dlib/src/dlib/linux/socket_linux.cpp
@@ -44,10 +44,13 @@ namespace dmSocket
         ifc.ifc_ifcu.ifcu_req = ifr;
         ifc.ifc_len = sizeof(buf);
         if (ioctl(s, SIOCGIFCONF, &ifc) < 0) {
+            dmLogInfo("<!> Unable to get interface addresses");
             close(s);
             return;
         }
 
+        dmLogInfo("<!> Got interface addresses");
+
         // NOTE: This is not compatible with BSD. You can't assume
         // equivalent size for all items
         int numif = ifc.ifc_len / sizeof(struct ifreq);
@@ -55,6 +58,7 @@ namespace dmSocket
             struct ifreq *r = &ifr[i];
 
             if (strcmp(r->ifr_name, "lo") == 0) {
+                dmLogInfo("<!> Skipping loopback interface");
                 continue;
             }
 
@@ -63,8 +67,10 @@ namespace dmSocket
 
             dmStrlCpy(a->m_Name, r->ifr_name, sizeof(a->m_Name));
 
-            if(ioctl(s, SIOCGIFADDR, r) < 0)
+            if(ioctl(s, SIOCGIFADDR, r) < 0) {
+                dmLogInfo("<!> Unable to get interface address");
                 continue;
+            }
 
             if (r->ifr_addr.sa_family == AF_INET)
             {
@@ -80,6 +86,10 @@ namespace dmSocket
                 a->m_Address.m_family = DOMAIN_IPV6;
                 memcpy(IPv6(&a->m_Address), &ia->sin6_addr, sizeof(struct in6_addr));
             }
+            else
+            {
+                dmLogInfo("<!> Unsupported address family %d", r->ifr_addr.sa_family);
+            }
 
             if(ioctl(s, SIOCGIFHWADDR, r) >= 0)
             {
@@ -91,9 +101,11 @@ namespace dmSocket
                 memset(a->m_MacAddress, 0x00, sizeof(unsigned char) * 6);
             }
 
-            if(ioctl(s, SIOCGIFFLAGS, r) < 0)
+            if(ioctl(s, SIOCGIFFLAGS, r) < 0) {
+                dmLogInfo("<!> Unable to get interface flags");
                 continue;
-            
+            }
+
             if (r->ifr_ifru.ifru_flags & IFF_UP) {
                 a->m_Flags |= FLAGS_UP;
             }
@@ -104,6 +116,8 @@ namespace dmSocket
             *count = *count + 1;
         }
 
+        dmLogInfo("<!> Got %d interface addresses", *count);
+
         close(s);
         return;
     }
diff --git a/engine/dlib/src/test/test_httpclient.cpp b/engine/dlib/src/test/test_httpclient.cpp
index ca4167c7a4..a6a2ca279a 100644
--- a/engine/dlib/src/test/test_httpclient.cpp
+++ b/engine/dlib/src/test/test_httpclient.cpp
@@ -541,6 +541,7 @@ TEST_P(dmHttpClientTest, NoKeepAlive)
 
     for (int i = 0; i < NUM_ITERATIONS; ++i)
     {
+        dmLogInfo("<!> TESTING NO-KEEP-ALIVE %d / %d", i, NUM_ITERATIONS);
         m_Content = "";
         dmSnPrintf(buf, sizeof(buf), "/no-keep-alive");
         dmHttpClient::Result r;
@@ -593,7 +594,7 @@ TEST_P(dmHttpClientTest, ClientTimeout)
     // We also want to keep the unit tests below a certain amount of seconds, so we also decrease the number of iterations in this loop.
 
     int sleep_time_ms = 5 * 1000;
-    #if defined(DM_PLATFORM_VENDOR) || defined (DM_SANITIZE_THREAD)
+    #if defined(DM_PLATFORM_VENDOR) || defined (DM_SANITIZE_THREAD) || defined(__aarch64__)
         const int timeout_us = 5000 * 1000;
         sleep_time_ms = 50 * 1000;
     #elif defined(__SCE__)
@@ -604,6 +605,12 @@ TEST_P(dmHttpClientTest, ClientTimeout)
 
     dmHttpClient::SetOptionInt(m_Client, dmHttpClient::OPTION_REQUEST_TIMEOUT, timeout_us); // microseconds
 
+#ifdef __aarch64__
+    dmLogInfo("<!!!> Platform is aarch64");
+#else
+    dmLogInfo("<!!!> Platform is x86_64");
+#endif
+
     char buf[128];
     for (int i = 0; i < 3; ++i)
     {
@@ -779,6 +786,25 @@ TEST_P(dmHttpClientTest, Test404)
     }
 }
 
+static std::string string_to_hex(const std::string& input)
+{
+    static const char hex_digits[] = "0123456789ABCDEF";
+
+    std::string output;
+    output.reserve(input.length() * 5);
+    output.push_back('{');
+    for (unsigned char c : input)
+    {
+        output.push_back('0');
+        output.push_back('x');
+        output.push_back(hex_digits[c >> 4]);
+        output.push_back(hex_digits[c & 15]);
+        output.push_back('+');
+    }
+    output.push_back('}');
+    return output;
+}
+
 TEST_P(dmHttpClientTest, Post)
 {
     for (int i = 0; i < 27; ++i)
@@ -788,15 +814,20 @@ TEST_P(dmHttpClientTest, Post)
         m_ToPost = "";
 
         for (int j = 0; j < n; ++j) {
-            char buf[2] = { (char)((rand() % 255) - 128), 0 };
+            int v = (rand() % 255) - 128;
+            char buf[2] = { (char)v, 0 };
             m_ToPost.append(buf);
-            sum += buf[0];
+            // dmLogInfo("<!> - %d + %d", sum, v);
+            sum += v;
         }
 
+        // dmLogInfo("<!> POSTing to %s://%s%s content %s", m_URI.m_Scheme, m_URI.m_Location, m_URI.m_Path, string_to_hex(m_ToPost).c_str());
+
         dmHttpClient::Result r;
         m_Content = "";
         m_StatusCode = -1;
         r = dmHttpClient::Post(m_Client, "/post");
+        dmLogInfo("<!> HTTP result is %d, expected sum %d => got %d", r, sum, atoi(m_Content.c_str()));
         ASSERT_EQ(dmHttpClient::RESULT_OK, r);
         ASSERT_EQ(200, m_StatusCode);
         ASSERT_EQ(sum, atoi(m_Content.c_str()));
@@ -813,9 +844,10 @@ TEST_P(dmHttpClientTest, PostLarge)
         m_ToPost = "";
 
         for (int j = 0; j < n; ++j) {
-            char buf[2] = { (char)((rand() % 255) - 128), 0 };
+            int v = (rand() % 255) - 128;
+            char buf[2] = { (char)v, 0 };
             m_ToPost.append(buf);
-            sum += buf[0];
+            sum += v;
         }
 
         dmHttpClient::Result r;
diff --git a/engine/dlib/wscript b/engine/dlib/wscript
index a64c762f4a..ca86d24d29 100644
--- a/engine/dlib/wscript
+++ b/engine/dlib/wscript
@@ -119,8 +119,11 @@ def shutdown(sht):
         print("Wrote test config file: %s" % configfilepath)
 
     try:
+        print("<!!!> RUNNING TESTS")
         waf_dynamo.run_tests(test_context, valgrind = True, configfile = configfilepath)
+        print("<!!!> TESTS FINISHED")
     finally:
+        print("<!!!> FINAL STEP - Removing test config file")
         os.remove(configfilepath)
 
     # The test should work on OSX, but the CI (Catalina) just crashed with no callstack (even in lldb)
diff --git a/engine/gamesys/src/gamesys/test/http/test_http_timeout.lua.raw b/engine/gamesys/src/gamesys/test/http/test_http_timeout.lua.raw
index 20be336dcb..2e7895a1a6 100644
--- a/engine/gamesys/src/gamesys/test/http/test_http_timeout.lua.raw
+++ b/engine/gamesys/src/gamesys/test/http/test_http_timeout.lua.raw
@@ -26,7 +26,7 @@ function test_http_timeout()
     options['timeout'] = 1.0
 
     print("Failing http requests ahead ->")
-    http.request("http://127.0.0.1:" .. PORT .. "/sleep/1.5", "GET",
+    http.request(ADDRESS .. "/sleep/1.5", "GET",
         function(response)
             assert(response.status == 0)
             requests_left = requests_left - 1
@@ -35,7 +35,7 @@ function test_http_timeout()
     requests_left = requests_left + 1
 
     -- The config file also specifies a timeout value, let's test that too
-    http.request("http://127.0.0.1:" .. PORT .. "/sleep", "GET",
+    http.request(ADDRESS .. "/sleep", "GET",
         function(response)
             assert(response.status == 0)
             requests_left = requests_left - 1
@@ -44,7 +44,7 @@ function test_http_timeout()
     requests_left = requests_left + 1
 
     -- Also test one that succeeds, i.e. does not time out
-    http.request("http://127.0.0.1:" .. PORT .. "/sleep/0.5", "GET",
+    http.request(ADDRESS .. "/sleep/0.5", "GET",
         function(response)
             assert(response.status == 200)
             requests_left = requests_left - 1
@@ -55,7 +55,7 @@ function test_http_timeout()
     for i=1,10 do
 
         options['timeout'] = 1.0
-        http.request(string.format("http://127.0.0.1:" .. PORT .. "/sleep/1.10%d", i), "GET",
+        http.request(string.format(ADDRESS .. "/sleep/1.10%d", i), "GET",
             function(response)
                 assert(response.status == 0) -- it should timeout
 
diff --git a/scripts/build.py b/scripts/build.py
index 8d3ec5ae34..4694137a4c 100755
--- a/scripts/build.py
+++ b/scripts/build.py
@@ -35,7 +35,7 @@ from threading import Thread, Event
 from queue import Queue
 from configparser import ConfigParser
 
-BASE_PLATFORMS = [  'x86_64-linux', #'arm64-linux',
+BASE_PLATFORMS = [  'x86_64-linux', 'arm64-linux',
                     'x86_64-macos', 'arm64-macos',
                     'win32', 'x86_64-win32',
                     'x86_64-ios', 'arm64-ios',
@@ -1010,6 +1010,9 @@ class Configuration(object):
         if self.target_platform in ('x86_64-macos','arm64-macos','arm64-ios','x86_64-ios') and 'linux' == sys.platform:
             strip = os.path.join(sdkfolder, 'linux', sdk.PACKAGES_LINUX_CLANG, 'bin', 'x86_64-apple-darwin19-strip')
 
+        if self.target_platform == 'arm64-linux' and self.host != self.target_platform:
+            strip = '/usr/bin/aarch64-linux-gnu-strip'
+
         run.shell_command("%s %s" % (strip, path))
         return True
 
@@ -1107,6 +1110,7 @@ class Configuration(object):
         supported_tests['x86_64-win32'] = ['win32', 'x86_64-win32', 'arm64-nx64', 'x86_64-ps4', 'x86_64-ps5']
         supported_tests['arm64-macos'] = ['x86_64-macos', 'arm64-macos', 'wasm-web', 'js-web']
         supported_tests['x86_64-macos'] = ['x86_64-macos', 'wasm-web', 'js-web']
+        supported_tests['x86_64-linux'] = ['arm64-linux']
 
         return self.target_platform in supported_tests.get(self.host, []) or self.host == self.target_platform
 
